packages/react-inbox/src/components/Messages2.0/Message.tsx

```diff
@@ -184,6 +187,14 @@ const MessageWrapper: React.FunctionComponent<
     }
   };
 
+  useOnScreen(messageHoverRef, () => {
+    if (opened || !messageId) {
+      return;
+    }
+
+    markMessageOpened(messageId);
```

Comment:
how do we handle failures or error state here?

Reply:
we don't really handle api failures at all.  i wonder what we could do better here